app/controllers/admin/categories_controller.rb

```diff
@@ -1,20 +1,44 @@
 class Admin::CategoriesController < AdminController
   def new
-    @categories = Category.new
+    @category = Category.new
   end
 
   def create
-
+    @category = Category.new(category_params)
+      if @category.save
+        flash[:notice] = 'A new category has been added.' 
+        redirect_to admin_categories_url
+      else
+        render 'new'
+      end
   end
   
   def index
-    @categories = Category.all.page(params[:page]).per(5)
+    @categories = Category.all.page(params[:page]).per(5).order(created_at: :desc)
   end
 
   def destroy
+    @category = Category.find(params[:id]).destroy
+    redirect_to admin_categories_url
   end
 
   def edit
-    @categories = Category.find(params[:id])
+    @category = Category.find(params[:id])
+  end
+
+  def update
+    @category = Category.find(params[:id])
+      if @category && @category.update_attributes(category_params)
```

Comment:
@ivanraybuglosa Can I know why `@category` is added in the condition? :)

Reply:
@inahsanjuan I usually use it to make sure that the id of the record is existing before I update.